### ISSUE TITLE
Could not find suitable distribution for Requirement.parse('pkgconfig')

### ISSUE THREAD

**Issue by baagaard-usgs** [geodynamics/pylith_installer#12]
Comment from Josimar on cig-short:

I am trying to build pylith-2.2.1 from source but the compilation stops at the h5py installation.  My machine runs on macOS v. 10.13.6. Note that I was able to build it on a linux computer without any problems.

A summary of the error is below and the complete full set log is attached. Would you have any suggestions to fix this issue ? 

Error summary:

```
Download error on https://pypi.org/simple/pkgconfig/: unknown url type: https -- Some packages may not be found!
Couldn't find index page for 'pkgconfig' (maybe misspelled?)
Download error on https://pypi.org/simple/: unknown url type: https -- Some packages may not be found!
No local packages or working download links found for pkgconfig
Traceback (most recent call last):
  File "setup.py", line 167, in <module>
    cmdclass = CMDCLASS,
  File "build/bdist.macosx-10.13-x86_64/egg/setuptools/__init__.py", line 128, in setup
  File "build/bdist.macosx-10.13-x86_64/egg/setuptools/__init__.py", line 123, in _install_setup_requires
  File "build/bdist.macosx-10.13-x86_64/egg/setuptools/dist.py", line 504, in fetch_build_eggs
  File "build/bdist.macosx-10.13-x86_64/egg/pkg_resources/__init__.py", line 774, in resolve
  File "build/bdist.macosx-10.13-x86_64/egg/pkg_resources/__init__.py", line 1057, in best_match
  File "build/bdist.macosx-10.13-x86_64/egg/pkg_resources/__init__.py", line 1069, in obtain
  File "build/bdist.macosx-10.13-x86_64/egg/setuptools/dist.py", line 571, in fetch_build_egg
  File "build/bdist.macosx-10.13-x86_64/egg/setuptools/command/easy_install.py", line 667, in easy_install
distutils.errors.DistutilsError: Could not find suitable distribution for Requirement.parse('pkgconfig')
make[2]: *** [h5py] Error 1
make[1]: *** [installed_h5py] Error 2
make: *** [installed_pylith] Error 2
```

**Comment by knepley** [comment]
Hmm, it looks like it is there: https://pypi.org/project/pkgconfig/

**Comment by lurpi** [comment]
Hi all, Luca here, I participated in the Pylith hackathon last year.
I have exactly this issue, it seems to be related to openSSL,  when I try to get pip manually by "python get-pip.py" I got the error

WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError("Can't connect to HTTPS URL because the SSL module is not available.",)': /simple/pip/

Setting the option  --enable-openssl for the installer does not help.

I am working on a virtual machine with Fedora24 (I can try on other, if needed)

**Comment by knepley** [comment]
This sounds like SSL is broken in your Python. Where did you get the Python?

**Comment by lurpi** [comment]
This was the Python coming with the original distribution (Python 2.7).

I am testing now with a clean Ubuntu distribution, without Python. 
I run the installer configure the with 
 --enable-python 
and 
--enable-openssl 
but I got to the same error (--enable-openssl seems to have no effect).

**Comment by baagaard-usgs** [comment]
There are two sets of options that should work:

1. Python and openssl from the system packages. This should work for Linux distributions!

2. Have the installer build **both** Python and openssl (`--enable-python --enable-openssl`). This should build openssl and the Python ssl modules.

**WARNING**: Having the installer build openssl (`--enable-openssl`) with the system Python will not provide Python with ssl modules, because the installer simply builds openssl, it doesn't build the relevant Python modules unless you have it build Python.

**Comment by lurpi** [comment]
Thanks for the clarification
With approach (2.), I keep on running into some issue with h5py and its installation (the error initially mentioned by Josimar in https://github.com/geodynamics/pylith_installer/issues/12#issue-447896092 ). I couldn't solve relying it on pylith installer only, with no python installed

Approach (1.) works if I install python-pip (as Matt mentioned

> This sounds like SSL is broken in your Python. Where did you get the Python?

I installed the most recent pip with a package manager: all dependencies (python included) were installed and that solved the issue for me (Ubuntu system, no previous python installation). On the Fedora system, with Python 2.7 and an old pip version, I was not able to update pip and I was stuck with the issue.  

Good to me, for the moment, thanks all.

**Comment by baagaard-usgs** [comment]
I believe this is resolved by PR #18 with the updates to how the installer uses OpenSSL along with all of the other clean up.
